src/shader_compiler_test.cc

```diff
@@ -216,4 +220,68 @@ TEST_F(ShaderCompilerTest, ReturnsCachedShader) {
   }
 }
 
+struct ParseSpvEnvCase {
+  std::string env_str;
+  bool ok;
+  uint32_t target_env;
+  uint32_t env_version;
+  uint32_t spirv_version;
+};
+
+using ParseSpvEnvTest = ::testing::TestWithParam<ParseSpvEnvCase>;
+
+TEST_P(ParseSpvEnvTest, Samples) {
+  uint32_t target_env = 42u;
+  uint32_t env_version = 43u;
+  uint32_t spirv_version = 44u;
+  auto r = amber::ParseSpvEnv(GetParam().env_str, &target_env, &env_version,
+                              &spirv_version);
+  if (GetParam().ok) {
+    EXPECT_TRUE(r.IsSuccess());
+    EXPECT_EQ(GetParam().target_env, target_env) << GetParam().env_str;
+    EXPECT_EQ(GetParam().env_version, env_version) << GetParam().env_str;
+    EXPECT_EQ(GetParam().spirv_version, spirv_version) << GetParam().env_str;
+  } else {
+    EXPECT_FALSE(r.IsSuccess());
+  }
+}
+
+// See also shaderc/env.h
+const uint32_t vulkan = 0;
+const uint32_t vulkan_1_0 = ((uint32_t(1) << 22));
+const uint32_t vulkan_1_1 = ((uint32_t(1) << 22) | (1 << 12));
+const uint32_t spv_1_0 = uint32_t(0x10000);
+const uint32_t spv_1_1 = uint32_t(0x10100);
+const uint32_t spv_1_2 = uint32_t(0x10200);
+const uint32_t spv_1_3 = uint32_t(0x10300);
+const uint32_t spv_1_4 = uint32_t(0x10400);
+
+INSTANTIATE_TEST_SUITE_P(ParseSpvEnvFailures,
+                         ParseSpvEnvTest,
+                         ::testing::ValuesIn(std::vector<ParseSpvEnvCase>{
```

Comment:
Since the tests are separate, would it be easier to just make to methods, one for the false and one for the true and remove the 2nd' param?

Reply:
Then I'd need to have two different test classes. I don't think that's worth the trouble.

Reply:
k